### PR TITLE
Change Start fx to Queue

### PR DESCRIPTION
**Pull request by jonathandiep**
Will be helpful to have these events as the indexer can utilize them

### CODE REVIEW

File: contracts/Quest.sol [hunk 1]
```diff
@@ -53,26 +53,31 @@ contract Quest is OwnableUpgradeable, IQuest {
...
     function unPause() external onlyOwner onlyStarted {
         isPaused = false;
+        emit Paused(false);
     }
 
     /// @notice Marks token ids as claimed
     /// @param tokenIds_ The token ids to mark as claimed
     function _setClaimed(uint256[] memory tokenIds_) private {
-        for (uint i = 0; i < tokenIds_.length;) {
+        for (uint i = 0; i < tokenIds_.length; ) {
```
Comment (JamieLottering) [inline contracts/Quest.sol:76]: Unnecessary trailing whitespace here after the semicolon?

File: contracts/Quest.sol [hunk 2]
```diff
@@ -105,11 +110,15 @@ contract Quest is OwnableUpgradeable, IQuest {
         if (tokens.length == 0) revert NoTokensToClaim();
 
         uint256 redeemableTokenCount = 0;
-        for (uint i = 0; i < tokens.length;) {
+        for (uint i = 0; i < tokens.length; ) {
```
Comment (JamieLottering) [inline contracts/Quest.sol:113]: Same here, intentional?

File: contracts/interfaces/IQuest.sol [hunk 3]
```diff
@@ -6,6 +6,8 @@ pragma solidity =0.8.16;
 interface IQuest {
     // This event is triggered whenever a call to #claim succeeds.
     event Claimed(address account, address rewardAddress, uint256 amount);
+    event Started(uint timestamp);
```
Comment (GarrettJMU) [inline contracts/interfaces/IQuest.sol:9]: ugh this is still confusing. Wondering if we rename this to queue

**Review by JamieLottering — APPROVED** [review]

**Comment by waynehoover** [comment]
I wonder if we should use the OZ Pausable contract, it includes events and it is a standard more or less: https://docs.openzeppelin.com/contracts/2.x/api/lifecycle#pausable

File: contracts/interfaces/IQuest.sol (hunk 3 above)
Comment (jonathandiep) [reply contracts/interfaces/IQuest.sol:9]: I'm cool with it, but should we rename just the event or the function as well?

File: contracts/Quest.sol (hunk 1 above)
Comment (jonathandiep) [reply contracts/Quest.sol:76]: It's intentional. I have this plugin formatting the solidity files: https://marketplace.visualstudio.com/items?itemName=NomicFoundation.hardhat-solidity

**Review by GarrettJMU — APPROVED** [review]

**Review by JamieLottering — APPROVED** [review]

File: contracts/Quest.sol [hunk 4]
```diff
@@ -20,7 +20,7 @@ contract Quest is PausableUpgradeable, OwnableUpgradeable, IQuest {
     uint256 public startTime;
     uint256 public totalParticipants;
     uint256 public rewardAmountInWeiOrTokenId;
-    bool public hasStarted;
+    bool public hasQueued;
```
Comment (waynehoover) [inline contracts/Quest.sol:23]: instead of `hasQueued` what about just `Queued`?

File: contracts/interfaces/IQuest.sol [hunk 5]
```diff
@@ -6,6 +6,7 @@ pragma solidity =0.8.16;
 interface IQuest {
     // This event is triggered whenever a call to #claim succeeds.
     event Claimed(address account, address rewardAddress, uint256 amount);
+    event Queued(uint timestamp);
```
Comment (waynehoover) [inline contracts/interfaces/IQuest.sol:9]: Whops just catching this now, but we need to add `indexed` to these params.

**Review by waynehoover — CHANGES_REQUESTED** [review]

**Review by waynehoover — APPROVED** [review]

**Pull request merged**